### PR TITLE
Simplify Branch constructor with parameter properties

### PR DESCRIPTION
**Pull request by Daniel-McCarthy**
Issue #4272 suggests to clean-up and refactor the `app/src/models/` files by making use of Typescript parameter properties. I have made changes to the `models/branch` file in order to reduce some redundancy via this feature.

The Issue also mentions recommending taking things one file at a time in regards to the `/models/` folder. If these changes are fitting then I would be happy to continue to another of the files after this pull request.

### CODE REVIEW

**Comment by shiftkey** [comment]
> If these changes are fitting then I would be happy to continue to another of the files after this pull request.

Works for me :tada:

**Comment by Daniel-McCarthy** [comment]
>Works for me

Wonderful. Would it be preferable to chain them in one pull request or split each into different pull requests?

Some are quite quick and simple, so I worry about adding too many pull requests to go through.

**Comment by iAmWillShepherd** [comment]
@Daniel-McCarthy doing them in separate PRs is fine, just be sure to state if one PR depends on another and you'll be good to go.

**Comment by shiftkey** [comment]
@Daniel-McCarthy I opened #5703 which updated the original refactoring example to use proper jsdoc following the discussion in https://github.com/desktop/desktop/pull/5697#issuecomment-423339885 and that seems to work fine. Are you able to update this PR to do the same?

**Comment by Daniel-McCarthy** [comment]
@shiftkey Certainly.
I am not sure if I have done it perfectly. Last night I set up VSCode to test this. When I used the JSDoc format for the `models/status` file it _seemed_ to be working as intended.

However, I noticed that the documentation showing up for the properties were showing the documentation from the super class, and that sub class did not seem to show any. This Branch file with the new changes seems to be behaving differently with documentation than the Status file was.

It could be behaving as expected, as I am not used to this environment.

File: app/src/models/branch.ts [hunk 1]
```diff
@@ -41,31 +41,21 @@ export function eligibleForFastForward(
   )
 }
 
-/** A branch as loaded from Git. */
+/**
+ * A branch as loaded from Git.
+ *
+ * @param name The short name of the branch. E.g., `master`.
```
Comment (shiftkey) [inline app/src/models/branch.ts]: I _think_ this should be documenting the `constructor` element below, not the other `class` itself.

Comment (shiftkey) [reply app/src/models/branch.ts]: @Daniel-McCarthy friendly bump on this change - it'd be nice to close this out before we open more PRs like this

Comment (Daniel-McCarthy) [reply app/src/models/branch.ts]: Ah, of course. Nice catch! Thanks for the feedback and reminding me of this.

**Pull request merged**